graph/src/components/versions/registry.rs

```diff
@@ -0,0 +1,87 @@
+use crate::prelude::FeatureFlag;
+use lazy_static::lazy_static;
+use semver::{Version, VersionReq};
+use std::collections::HashMap;
+
+lazy_static! {
+    pub static ref VERSIONS: HashMap<Version, Vec<FeatureFlag>> = {
+        let mut supported_versions: Vec<(Version, Vec<FeatureFlag>)> = vec![
+            // baseline version
+            (Version::new(1, 0, 0), vec![]),
+            // Versions with feature flags
+            (Version::new(1, 1, 0), vec![FeatureFlag::BasicOrdering])
+        ];
+
+        let mut map = HashMap::new();
+
+        // Sort version by major, minor, patch, from higher to lower.
+        supported_versions.sort_by(|a, b| b.0.partial_cmp(&a.0).unwrap());
+
+        for (version, flags) in supported_versions {
+            map.insert(version, flags);
+        }
+
+        map
+    };
+
+    static ref LATEST_VERSION: String = {
+        let keys: Vec<Version> = VERSIONS.clone().into_keys().collect();
+
+        // Versions are sorted by major, minor, patch, from higher to lower.
+        keys.first().unwrap().to_string()
```

Comment:
This is wrong `HashMap.into_keys` will return keys in arbitrary order. `BTreeMap` would work, but I think `first` would return the smallest version, so either reverse the iterator or put the keys into a `Vec` and sort that

Reply:
I did a refactor and it's now using `VERSION_COLLECTION: HashMap<Version, Vec<FeatureFlag>>` to get the feature flags and `VERSIONS: Vec<&'static Version>` (sorted from latest to oldest version, higher -> lower).

The `VERSIONS` vector is sorted because of the logic in `resolve()` function.